modules/consent/lib/Auth/Process/Consent.php

```diff
@@ -116,7 +116,15 @@ public function __construct($config, $reserved)
             $this->_hiddenAttributes = $config['hiddenAttributes'];
         }
 
-        if (array_key_exists('noconsentattributes', $config)) {
+        if (array_key_exists('attributes.exclude', $config)) {
+            if (!is_array($config['attributes.exclude'])) {
+                throw new SimpleSAML_Error_Exception(
+                    'Consent: attributes.exclude must be an array. '.
+                    var_export($config['attributes.exclude'], true).' given.'
+                );
+            }
```

Comment:
Nothing is actually set/saved in this case?

Reply:
Well spotted! Seems I lost a line when I copied over the original diff.

Reply:
Given that the module already has some test cases, would it be easy to add two for the two different settings so something like this would have been caught?

Reply:
Looking at exactly that right now :)

Reply:
There doesn't seem to be an easy way to test the functionality within process() - at least not without some substantive changes, because it issues a redirect and we'd need to examine the state after that completes (the problem with exit). But I've just added one that tests the correct setting of the instance private variables using a ReflectionClass.